forms/collections/pages.js

```diff
@@ -19,9 +19,15 @@ define(function (require) {
       var self, currentform, currentPage;
       var view;
       self = this;
+
       if (!_.isNumber(index)) {
         index = 0;
       }
+
+      if (self.current) {
+        self.current.removeView();
+      }
+
```

Comment:
I wonder if it's safer to just move the "delete other page views" loop from below to this spot, and make it delete all page views? Instead of only deleting 1 page here?